lib/rmg/CRmgTemplate.h

```diff
@@ -41,17 +41,24 @@ namespace EWaterContent
 	};
 }
 
+namespace EZoneMonsterStrength
```

Comment:
Why not an enum class?

Reply:
Because all enums around are also enums-in-namespaces and not enum classes, so I guess for consistency it might be better if they're either all enum classes or all enums-in-namespaces. And we use those enums as integers. For example global monster strength "strong" corresponds to number 4, which goes up by 1 in zones with "strong" monsters and that 5 is then used to fetch the correct coefficients from arrays in guard count calculations. Not sure how that part of code would look with enum classes. I'm referring to this (in \lib\rmg\ObjectManager.cpp, around line 400):
```
CGCreature * ObjectManager::chooseGuard(si32 strength, bool zoneGuard)
{
	//precalculate actual (randomized) monster strength based on this post
	//http://forum.vcmi.eu/viewtopic.php?p=12426#12426
	
	int mapMonsterStrength = map.getMapGenOptions().getMonsterStrength();
	int monsterStrength = (zoneGuard ? 0 : zone.monsterStrength - EZoneMonsterStrength::NORMAL) + mapMonsterStrength - 1; //array index from 0 to 4
	static const std::array<int, 5> value1{2500, 1500, 1000, 500, 0};
	static const std::array<int, 5> value2{7500, 7500, 7500, 5000, 5000};
	static const std::array<float, 5> multiplier1{0.5, 0.75, 1.0, 1.5, 1.5};
	static const std::array<float, 5> multiplier2{0.5, 0.75, 1.0, 1.0, 1.5};
	
	int strength1 = static_cast<int>(std::max(0.f, (strength - value1.at(monsterStrength)) * multiplier1.at(monsterStrength)));
	int strength2 = static_cast<int>(std::max(0.f, (strength - value2.at(monsterStrength)) * multiplier2.at(monsterStrength)));
	
	strength = strength1 + strength2;
```

Reply:
@IvanSavenko what do you think?

Reply:
Don't know?

Yes, enum class is preferable, except for cases when enum is often used as integer type, making enum class too inconvenient.
Whether these use-cases here qualify as "often used" - not sure. Not familiar enough with RMG code.

Reply:
These 2 enums appear as numbers in monster generation formula and the global one is also shown to the player as an int in the map description of the generated random map.
EZoneMonsterStrength has 2 uses: parsing the `"monsters" : "normal", //"weak" "strong"` part of the rmg template json written by the template makers and then using that to increase or decrease monster strength in that zone.
However, all the other enums in rmg don't seem to have strong ties to ints, but I will fix that in a different PR as such fix wouldn't match this PR or branch name.

Reply:
Correction: scenario description in a generated random map in game shows `monsters strong`, not `monsters 4`. 'monsters 4' is what HotA and SoD shows.